### PR TITLE
Add GuildMember#pending

### PR DESCRIPTION
**Pull request by wasdennnoch**
**Please describe the changes this PR makes and why it should be merged:**

https://github.com/discord/discord-api-docs/pull/2299
Previously (when it was still undocumented) that field was nullable, the new merged docs now say it's always present; however someone already noted that this field is not actually always present. Will update once that has been clarified.

**Status**

- [x] Code changes have been tested against the Discord API, or there are no code changes
- [x] I know how to update typings and have done so, or typings don't need updating

**Semantic versioning classification:**

- [x] This PR changes the library's interface (methods or parameters added)
  - [ ] This PR includes breaking changes (methods removed or renamed, parameters moved or removed)
- [ ] This PR **only** includes non-code changes, like changes to documentation, README, etc.

### CODE REVIEW

**Comment by wasdennnoch** [comment]
Docs have been clarified that `pending` isn't nullable anymore (https://github.com/discord/discord-api-docs/commit/2fe4e62a86f1d6b79bb57e3b46f0016b03391261) and may not always be present, in which case the member isn't supposed to be pending anyways. Updated the PR accordingly.

**Review by Rabustep — APPROVED** [review]

File: src/structures/GuildMember.js [hunk 1]
```diff
@@ -83,6 +89,7 @@ class GuildMember extends Base {
     if ('joined_at' in data) this.joinedTimestamp = new Date(data.joined_at).getTime();
     if ('premium_since' in data) this.premiumSinceTimestamp = new Date(data.premium_since).getTime();
     if ('roles' in data) this._roles = data.roles;
+    if ('pending' in data) this.pending = data.pending;
```
Comment (vladfrangu) [inline src/structures/GuildMember.js:92]: Can't we just do `this.pending = data.pending ?? false`? cc @kyranet

Comment (kyranet) [reply src/structures/GuildMember.js:92]: We can, as a matter of a fact, #5132 makes all the lines above use `??`.

**Review by SpaceEEC — APPROVED** [review]

**Review by vladfrangu — CHANGES_REQUESTED** [review]
See comment before, use ??

File: src/structures/GuildMember.js (hunk 1 above)
Comment (wasdennnoch) [reply src/structures/GuildMember.js]: Done

**Review by kyranet — APPROVED** [review]

**Review by vladfrangu — APPROVED** [review]

**Review by iCrawl — APPROVED** [review]

**Pull request merged**